src/index.ts

```diff
@@ -203,6 +201,9 @@ class MagicNixCacheAction {
     actionsCore.debug("Full daemon start command:");
     actionsCore.debug(`${daemonBin} ${daemonCliFlags.join(" ")}`);
 
+    this.daemonStarted = true;
+    actionsCore.saveState(STATE_STARTED, STARTED_HINT);
+
     // Start the server. Once it is ready, it will notify us via the notification server.
     const daemon = spawn(daemonBin, daemonCliFlags, opts);
```

Comment:
Why don't we just move it after we spawn the daemon? That seems like a more correct time to say it's been started.

Or maybe even right after we log "Launched Magic Nix Cache" after waiting for it to start...?